Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -184,10 +192,20 @@ public actor MemcachedConnection {
             if var unwrappedResponse = response {
                 return Value.readFromBuffer(&unwrappedResponse)
             } else {
-                throw MemcachedConnectionError.unexpectedNilResponse
+                throw MemcachedError(
```

Comment:
Not for all of these methods e.g. `get`, `set` etc. We don't have to do a state check, we can just call `self.sendRequest` which is doing the check anyhow. That should make our implementation here a lot shorter.